app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt

```diff
@@ -88,89 +96,140 @@ fun SplashScreen(
+        if (shouldShowLoading) Loading()
+        errorState?.let { error ->
+            ErrorAlertDialog(
+                errorState = error,
+                title = stringResource(id = R.string.oops),
+                buttonText = stringResource(id = android.R.string.ok),
+                onClickButton = { viewModel.resetError() }
+            )
         }
     }
     viewModel.startTimerToNavigateToLogin(splashTime = splashTime)
 }
 
-@Suppress("LongMethod")
 @Composable
-fun LoginComponents() {
+fun BackgroundImage(modifier: Modifier, isBlurred: Boolean) {
+    Image(
+        painter = painterResource(id = if (isBlurred) R.drawable.ic_overlay else R.drawable.splash_bg),
+        contentDescription = stringResource(id = R.string.splash_background_content_description),
+        modifier = modifier,
+        contentScale = ContentScale.Crop
+    )
+}
+
+@Composable
+fun LogoImage(modifier: Modifier) {
+    Image(
+        painter = painterResource(id = R.drawable.ic_logo_white),
+        contentDescription = stringResource(id = R.string.splash_logo_content_description),
+        modifier = modifier
+    )
+}
+
+@Composable
+fun LoginComponents(viewModel: SplashViewModel = hiltViewModel()) {
     Column(verticalArrangement = Arrangement.spacedBy(20.dp)) {
         var enableLoginButton by remember { mutableStateOf(false) }
-        var emailState by remember { mutableStateOf("") }
-        var passwordState by remember { mutableStateOf("") }
+        var email by remember { mutableStateOf("") }
+        var password by remember { mutableStateOf("") }
 
         EmailTextField(
-            emailState = emailState,
+            emailState = email,
             onValueChange = {
-                emailState = it
-                enableLoginButton = emailState.isNotEmpty() && passwordState.isNotEmpty()
+                email = it
+                enableLoginButton = email.isNotEmpty() && password.isNotEmpty()
             }
         )
         PasswordTextField(
-            passwordState = passwordState,
+            passwordState = password,
             onValueChange = {
-                passwordState = it
-                enableLoginButton = emailState.isNotEmpty() && passwordState.isNotEmpty()
+                password = it
+                enableLoginButton = email.isNotEmpty() && password.isNotEmpty()
             }
         )
-        LoginButton(loginButtonState = enableLoginButton)
+        LoginButton(loginButtonState = enableLoginButton) {
+            viewModel.login(email = email, password = password)
+        }
     }
 }
 
 @Composable
 fun EmailTextField(emailState: String, onValueChange: (String) -> Unit) {
+    val focusManager = LocalFocusManager.current
+    val emailContentDescription = stringResource(id = R.string.login_email_text_field)
+
     TextField(
         value = emailState,
         onValueChange = { onValueChange(it) },
         shape = RoundedCornerShape(12.dp),
```

Comment:
`12.dp` is used pretty frequently for corner radius. How about we extract it to a new `Dimens.kt` folder inside `ui/theme`?
```kotlin
// Dimens.kt
val CornerRadius = 12.dp
```
```suggestion
        shape = RoundedCornerShape(CornerRadius),
```